### PR TITLE
added solution to the project:

### PR DESCRIPTION
**Pull request by nikolasdymchenko7**
- Bulldozer, Excavator, Truck classes -> new special fields, new constructor, getter and setter methods for the fields
- parameterized interface and methods
- created new 3 classes BulldozerProducer, ExcavatorProducer, TruckProducer with implementation interface MachineProducer
- implementation interface MachineService to the class MachineServiceImpl

### CODE REVIEW

File: src/main/java/core/mate/academy/service/BulldozerProducer.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Bulldozer;
+import java.util.List;
+
+public class BulldozerProducer implements MachineProducer {
+
```
Comment (Rommelua) [inline src/main/java/core/mate/academy/service/BulldozerProducer.java:7]: empty line

File: src/main/java/core/mate/academy/service/BulldozerProducer.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Bulldozer;
+import java.util.List;
+
+public class BulldozerProducer implements MachineProducer {
```
Comment (Rommelua) [inline src/main/java/core/mate/academy/service/BulldozerProducer.java:6]: do not use raw types

File: src/main/java/core/mate/academy/service/ExcavatorProducer.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Excavator;
+import java.util.List;
+
+public class ExcavatorProducer implements MachineProducer {
```
Comment (Rommelua) [inline src/main/java/core/mate/academy/service/ExcavatorProducer.java:6]: do not use raw types

File: src/main/java/core/mate/academy/service/ExcavatorProducer.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
```
Comment (Rommelua) [inline src/main/java/core/mate/academy/service/ExcavatorProducer.java:7]: empty line

File: src/main/java/core/mate/academy/service/MachineService.java [hunk 5]
```diff
@@ -26,13 +27,13 @@ public interface MachineService {
...
+    void fill(List<? super Machine> machines, T value);
 
     /**
      * Call the method doWork() from each machine.
      * This method should be able to accept a list of bulldozers as well as list of trucks.
      *
      * @param machines - the list of machines
      */
-    void startWorking(List<Object> machines);
+    void startWorking(List<? extends Machine> machines);
```
Comment (Rommelua) [inline src/main/java/core/mate/academy/service/MachineService.java:38]: use T

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 6]
```diff
@@ -1,7 +1,43 @@
...
+    private ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+    private TruckProducer truckProducer = new TruckProducer();
+
+    @Override
+    public List getAll(Class type) {
+        if (type == Bulldozer.class) {
+            return new ArrayList(bulldozerProducer.get());
+        } else if (type == Excavator.class) {
+            return new ArrayList(excavatorProducer.get());
+        } else if (type == Truck.class) {
```
Comment (Rommelua) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:24]:
```suggestion
        } if (type == Excavator.class) {
            return new ArrayList(excavatorProducer.get());
        } if (type == Truck.class) {
```

File: src/main/java/core/mate/academy/service/TruckProducer.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Truck;
+import java.util.List;
+
+public class TruckProducer implements MachineProducer {
+
+    @Override
```
Comment (Rommelua) [inline src/main/java/core/mate/academy/service/TruckProducer.java:8]: same

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 8]
```diff
@@ -1,7 +1,45 @@
...
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    private BulldozerProducer bulldozerProducer = new BulldozerProducer();
+    private ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+    private TruckProducer truckProducer = new TruckProducer();
```
Comment (aaananas) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:16]:
```suggestion
    private MachineProducer<Bulldozer> bulldozerProducer = new BulldozerProducer();
    private MachineProducer<Excavator> excavatorProducer = new ExcavatorProducer();
    private MachineProducer<Truck>truckProducer = new TruckProducer();
```

**Review by aaananas — APPROVED** [review]
remember about abstraction

**Review by aaananas — APPROVED** [review]

**Pull request closed without merging**